### ISSUE TITLE
conda install name=version=build does not install dependencies

### ISSUE THREAD

**Issue by ilanschnell** [conda/conda#2066]
Using `conda 3.19.1`, I see this strange behavior (Linux 64, but that does not matter):
```
$ conda install graphviz=2.38.0=1
Fetching package metadata: ....
Solving package specifications: 
Package plan for installation in environment /home/ilan/minonda:

The following NEW packages will be INSTALLED:

    graphviz: 2.38.0-1
```
Where as is I just say `conda install graphviz=2.38.0` or `conda install graphviz`, conda will also install the dependencies:
```
$ conda install graphviz
Fetching package metadata: ....
Solving package specifications: .................
Package plan for installation in environment /home/ilan/minonda:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    glib-2.43.0                |                0         6.4 MB
    harfbuzz-0.9.39            |                0         1.1 MB
    pango-1.39.0               |                0         777 KB
    ------------------------------------------------------------
                                           Total:         8.2 MB

The following NEW packages will be INSTALLED:

    cairo:      1.12.18-6
    fontconfig: 2.11.1-5 
    glib:       2.43.0-0 
    graphviz:   2.38.0-1 
    harfbuzz:   0.9.39-0 
    libxml2:    2.9.2-0  
    pango:      1.39.0-0 

The following packages will be UPDATED:

    freetype:   2.5.2-2  --> 2.5.5-0  
    pixman:     0.26.2-0 --> 0.32.6-0
```

**Comment by mcg1969** [comment]
This isn't a bug, it's a behavior change. The New And Improved Conda (TM) should never be allowed to result in a broken/inconsistent state unless you use the `--force` flag. How do we know, after all, that the user _doesn't_ want to install dependencies in this case?

**Comment by ilanschnell** [comment]
@mcg1969 But in this case, the user is just being very explicit about the package they want to install, and is not even using the `--force` option, and they get a broken package (because the dependencies are missing).

**Comment by mcg1969** [comment]
Sorry Ilan, I misread your post. We do indeed need to make sure that the dependencies are installed here. The old version of `conda` skips dependencies in this case because it thinks it is an explicit spec.

**Comment by groutr** [comment]
IMO, dependencies always need to be resolved, even if it is an explicit spec.  I believe `--force` could even use the solver (by instructing the solver to do whatever needs to make a package installed).  If you really need to disable the solver, you can use an option like `--no-deps`.  Implicitly bypassing the solver produces many confused users.

**Comment by kalefranz** [comment]
@mcg1969 is this fixed with `4.0`?

**Comment by mcg1969** [comment]
Should be, yes.
